### PR TITLE
feat: added builtin vars support

### PR DESCRIPTION
**Pull request by capri-xiyue**
1. added builtin vars support
2. added test case with no default spec yaml

### CODE REVIEW

File: templates/common/builtinvar/builtin_vars.go [hunk 1]
```diff
@@ -68,3 +68,15 @@ func NamesInScope(f features.Features) []string {
 
 	return out
 }
+
+// ValidateBuiltinVars returns error if any of the attemptedNames are not valid builtin
+// var names.
+func ValidateBuiltinVars(attemptedNames []string) error {
```
Comment (drevell) [inline templates/common/builtinvar/builtin_vars.go:74]: The set of builtin vars depends on the api_version, so we can't hardcode it like this. But the good news is that the function that we need already exists, it's the `Validate()` function on line 49 of this file.

**Review by drevell — CHANGES_REQUESTED** [review]

File: templates/common/builtinvar/builtin_vars.go (hunk 1 above)
Comment (capri-xiyue) [reply templates/common/builtinvar/builtin_vars.go]: done

File: templates/commands/goldentest/new_test_cli.go [hunk 2]
```diff
@@ -99,7 +100,16 @@ func (c *NewTestCommand) Run(ctx context.Context, args []string) (rErr error) {
 		return err //nolint:wrapcheck
 	}
 
-	buf, err := marshalTestCase(resolvedInputs)
+	builtinVarsKeys := make([]string, 0, len(c.flags.BuiltinVars))
+	for k := range c.flags.BuiltinVars {
+		builtinVarsKeys = append(builtinVarsKeys, k)
+	}
+	err = builtinvar.Validate(spec.Features, builtinVarsKeys)
+	if err != nil {
```
Comment (sethvargo) [inline templates/commands/goldentest/new_test_cli.go:108]: Nit: cuddle errors.

Comment (capri-xiyue) [reply templates/commands/goldentest/new_test_cli.go:108]: fixed by https://github.com/abcxyz/abc/pull/388

**Pull request merged**